src/main/java/org/craftercms/studio/impl/v2/service/content/internal/ContentTypeServiceInternalImpl.java

```diff
@@ -199,13 +290,324 @@ public String getContentTypeTemplatePath(String siteId, String contentTypeId) th
+
+        if (CollectionUtils.isNotEmpty(allContentTypes)) {
+            List<ContentTypeConfigTO> contentTypes = new ArrayList<>();
+            for (ContentTypeConfigTO contentTypeConfig : allContentTypes) {
+                // check if the path matches one of includes paths
+                if (CollectionUtils.isNotEmpty(contentTypeConfig.getPathIncludes())) {
+                    for (String pathIncludes : contentTypeConfig.getPathIncludes()) {
+                        if (relativePath.matches(pathIncludes)) {
+                            logger.trace("In site '{}' path '{}' matches '{}'", siteId, relativePath, pathIncludes);
+                            boolean isMatch = true;
+                            if (contentTypeConfig.getPathExcludes() != null) {
+                                for (String excludePath : contentTypeConfig.getPathExcludes()) {
+                                    if (relativePath.matches(excludePath)) {
+                                        logger.trace("In site '{}' path '{}' matches an exclude path '{}'",
+                                                siteId, relativePath, excludePath);
+                                        isMatch = false;
+                                        break;
+                                    }
+                                }
+                            }
+                            if (isMatch) {
+                                // if a match is found, populate the content type information
+                                addContentTypes(userRoles, contentTypeConfig, contentTypes);
+                            }
+                        }
+                    }
+                } else if (CollectionUtils.isEmpty(contentTypeConfig.getPathExcludes())) {
+                    addContentTypes(userRoles, contentTypeConfig, contentTypes);
+                }
+            }
+            return contentTypes;
+        } else {
+            logger.error("No content type path configuration is found for site '{}'", siteId);
+            return null;
+        }
+    }
+
+    /**
+     * Add a content type to a list if allowed
+     * @param userRoles list of user roles
+     * @param config {@link ContentTypeConfigTO} object to check
+     * @param contentTypes list of {@link ContentTypeConfigTO} to add
+     */
+    protected void addContentTypes(Set<String> userRoles, ContentTypeConfigTO config,
+                                   List<ContentTypeConfigTO> contentTypes) {
+        boolean isAllowed = this.isUserAllowed(userRoles, config);
+        if (isAllowed) {
+            contentTypes.add(config);
+        }
+    }
+
+    /**
+     * Check if user is allowed to access given content type
+     * @param userRoles list of user roles
+     * @param item {@link ContentTypeConfigTO} object to check
+     * @return true if allowed, false otherwise
+     */
+    protected boolean isUserAllowed(Set<String> userRoles, ContentTypeConfigTO item) {
```

Comment:
@sumerjabri should we consider updating this functionality to check for permissions instead of roles?